src/firefly/java/edu/caltech/ipac/firefly/server/dpanalyze/DataProductAnalyzer.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * License information at https://github.com/Caltech-IPAC/firefly/blob/master/License.txt
+ */
+package edu.caltech.ipac.firefly.server.dpanalyze;
+
+import edu.caltech.ipac.firefly.core.FileAnalysisReport;
+import nom.tam.fits.Header;
+
+import java.io.File;
+import java.util.Map;
+
+/**
+ * @author Trey Roby
+ */
+public interface DataProductAnalyzer {
+    default FileAnalysisReport analyze(FileAnalysisReport inputReport,
+                                       File inFile,
+                                       String analyzerId,
+                                       Map<String,String> params) { return inputReport;}
+    default FileAnalysisReport analyzeFits(FileAnalysisReport inputReport,
```

Comment:
Unfortunately, to be able to operate on a FITS you need `Header[]`. It would be nice to be more generic but I don't see how it could be done. There are 2 issues:

1. The FileAnalyzer is not passing this data blind. Its knows it has a FITS and it already has the `Header[]` so being able to call an interface the can handle the FITS data without having to re-read is preferable. This could be a serious performance issue.
 
1. Opening and closing FITS is problematic. Our code has yet to get it right the first time. I found that the FileAnalyzer was leaving all its FITS file opened. I would prefer not having the interfaces constantly opening and closing FITS files. 

_Note_- It would be possible to put each header in the `inputReport`, but you would have the same issue. a FITS specific structure in the input report.

Reply:
We could use the raw interface [FitsElement](https://github.com/nom-tam-fits/nom-tam-fits/blob/master/src/main/java/nom/tam/fits/FitsElement.java) which is implemented by lot of data structure we could support the same way with FITS library. 

> All Known Implementing Classes:
AbstractTableData, AsciiTable, AsciiTableHDU, BasicHDU, BinaryTable, BinaryTableHDU, CompressedImageData, CompressedImageHDU, Data, FitsHeap, Header, ImageData, ImageHDU, RandomGroupsData, RandomGroupsHDU, TableHDU, UndefinedData, UndefinedHDU

Also, i don't think reading header needs to read the full file so opening and closing I/O should be fast, isn't ? unless it has to be done thousand of times a minute but this operation would happen on highlight row only for example most, correct?